aiida/storage/psql_dos/orm/querybuilder/main.py

```diff
@@ -138,78 +138,65 @@ def table_groups_nodes(self):
         return aiida.storage.psql_dos.models.group.table_groups_nodes
 
     def get_session(self) -> Session:
-        """
-        :returns: a valid session, an instance of :class:`sqlalchemy.orm.session.Session`
-        """
+        """Get the connection to the database"""
         return self._backend.get_session()
 
     def count(self, data: QueryDictType) -> int:
-        with self.use_query(data) as query:
-            result = query.count()
+        with self.query_session(data) as build:
+            result = build.query.count()
         return result
 
     def first(self, data: QueryDictType) -> Optional[List[Any]]:
-        with self.use_query(data) as query:
-            result = query.first()
+        with self.query_session(data) as build:
+            result = build.query.first()
 
         if result is None:
             return result
 
-        # we discard the first item of the result row,
-        # which was what the query was initialised with and not one of the requested projection (see self._build)
-        result = result[1:]
-
-        if len(result) != self._requested_projections:
-            raise AssertionError(
-                f'length of query result ({len(result)}) does not match '
-                f'the number of specified projections ({self._requested_projections})'
-            )
+        # SQLA will return a Row, if only certain columns are requested,
+        # or a database model if that is requested
+        if not isinstance(result, Row):
+            result = (result,)
```

Comment:
This is the only part I was a little unsure of; since there can now be only a single projection (as opposed to >=2 before), the return type can be different, whether you are returning a whole row e.g. `Node` or a single column e.g `Node.id`.
Without this, the tests were failing, but with it, they appear fine 🤷